src/me/mrCookieSlime/Slimefun/Objects/SlimefunItem/SlimefunMachine.java

```diff
@@ -80,5 +99,34 @@ public MultiBlock toMultiBlock() {
 	public Iterator<ItemStack[]> recipeIterator() {
 		return this.recipes.iterator();
 	}
-
+	
+	// Overloaded method for finding a potential output chest. Fallbacks to the old system of putting the adding back into the dispenser.
+	// Optional last argument Inventory placeCheckerInv is for multiblock machines that create a dummy inventory to check if there's a space for the adding,
+	// i.e. Enhanced crafting table
+	public static Inventory findValidOutputInv(ItemStack adding, Block dispBlock, Inventory dispInv) {
+		return findValidOutputInv(adding, dispBlock, dispInv, dispInv);
+	}
+	
+	public static Inventory findValidOutputInv(ItemStack product, Block dispBlock, Inventory dispInv, Inventory placeCheckerInv) {
+		Inventory outputInv = null;
+		for (BlockFace face : faces) {
+			Block potentialOutput = dispBlock.getRelative(face);
+			if (BlockStorage.hasBlockInfo(potentialOutput) && BlockStorage.checkID(potentialOutput).equals("OUTPUT_CHEST")) {
```

Comment:
This may be heavy call, you should probably invoke checkID once, store the String and then do a null-check and an equals check instead.
It would greatly increase the performance.

Reply:
Is this better?

if (id != null && id.equals("OUTPUT_CHEST")) {

Reply:
Yes.